std/algorithm/comparison.d

```diff
@@ -1567,3 +1567,156 @@ unittest
                 2, "two",
                 ));
 }
+
+/** Returns: First Argument (element of $(D a)) whose implicit conversion to
+    bool is true.
+
+    Similar to behaviour of or operator in dynamic languages such as Lisp's (or
+    a...) and Python's a or ....
+
+    TODO Is inout Conversion!T the correct return value?
+
+    NOTE: Lazy parameters are currently marked as throw which makes it
+    impossible to mark either() as nothrow. Issue at
+    https://issues.dlang.org/show_bug.cgi?id=12647
+*/
+CommonType!Ts either(Ts...)(lazy Ts a) if (a.length >= 1)
```

Comment:
Nice, two things:

1. Could you please try whether it's possible to accept/return by reference? I.e. `either(a[0], a[1]) = 1;` should work. But I don't think auto ref cooperates nicely with lazy.

2. A predicate would help, e.g. choose either non-empty string: `auto s = either!(s => !s.empty)(param, "default");`. In fact I think such use would be most useful.